engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/canonicalgraph/CanonicalProcess.scala

```diff
@@ -10,41 +10,53 @@ sealed trait CanonicalTreeNode
 
 object CanonicalProcess {
 
-  def withoutDisabledNodes(process: CanonicalProcess): CanonicalProcess = {
-    def withoutDisabled(nodes: List[CanonicalNode]): List[CanonicalNode] = {
-      nodes.filter {
-        _.data match {
-          case nodeData: Disableable if nodeData.isDisabled.contains(true) => false
-          case _ => true
-        }
-      }.map {
-        case flatNode: canonicalnode.FlatNode =>
-          flatNode
-
-        case filter: canonicalnode.FilterNode =>
-          filter.copy(nextFalse = withoutDisabled(filter.nextFalse))
-
-        case switch: canonicalnode.SwitchNode =>
-          switch.copy(
-            nexts = switch.nexts.map { caseNode =>
-              caseNode.copy(nodes = withoutDisabled(caseNode.nodes))
-            }.filterNot(_.nodes.isEmpty),
-            defaultNext = withoutDisabled(switch.defaultNext))
-
-        case split: canonicalnode.SplitNode =>
-          split.copy(nexts = split.nexts.map(withoutDisabled).filterNot(_.isEmpty))
-
-        case subprocess: canonicalnode.Subprocess =>
-          subprocess.copy(
-            outputs = subprocess.outputs.map { case (id, canonicalNodes) =>
-              (id, withoutDisabled(canonicalNodes))
-            }.filterNot { case (_, canonicalNodes) => canonicalNodes.isEmpty }
+  private def withoutDisabled(nodes: List[CanonicalNode]): List[CanonicalNode] =
+    nodes.foldLeft(List.empty[CanonicalNode]) { (agg, node) =>
```

Comment:
It looks like it is simple transformation 1 to 0...n, why not to just use `flatMap`?

Reply:
:+1: